index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2019-present, ProductBoard, Inc.
+ * All rights reserved.
+ */
+
+const fs = require("fs");
+const child = require("child_process");
+const gzipSize = require("gzip-size");
+
+/**
+ * Webpack Plugin for generating info JSON (hash, sha, assetSize), it's consumable by webpack-deploy later on when deploying
+ */
+class BuildInfo {
+  constructor({ filename = "build.log" }) {
+    this.filename = filename;
+  }
+
+  getSha() {
+    return child
+      .execSync("git rev-parse HEAD")
+      .toString()
+      .trim()
+      .slice(0, 7);
```

Comment:
We have a config value that decides the length of SHAs. 7 proved to be not enough even for us. In general I would advise to provide the full SHA and leave the format up to the consumer.